images/generic/src/image.rs

```diff
@@ -0,0 +1,68 @@
+use tc_core::{Container, Docker, Image, WaitError, WaitForMessage};
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum WaitFor {
+    Nothing,
+    LogMessage(String),
+}
+
+impl WaitFor {
+    fn wait<D: Docker, I: Image>(&self, container: &Container<D, I>) -> Result<(), WaitError> {
+        match self {
+            WaitFor::Nothing => Ok(()),
+            WaitFor::LogMessage(message) => container.logs().stderr.wait_for_message(message),
```

Comment:
Not sure if hardcoding `stderr` here is a good idea.

The other thing is, a user would currently have to call `to_owned` on a static string but most times, what they will pass is a `&'static str`.

Maybe we can kill two birds with one stone here and create factory functions on `WaitFor` to initialize it from a `&str` and at the same time configure a flag, which stream the message is expected on.

```rust
impl WaitFor {

    fn message_on_stdout(message: &str) -> Self {
        Self::LogMessage {
            message: message.to_owned(),
            stream: Stream::StdOut
        }
    }


    fn message_on_stderr(message: &str) -> Self {
        Self::LogMessage {
            message: message.to_owned(),
            stream: Stream::StdErr
        }
    }
}
```

Reply:
@thomaseizinger  
Fixed